addons/survey/static/src/js/survey_form.js

```diff
@@ -469,7 +470,29 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
                 }
             }
 
-            this._prepareSubmitValues(formData, params);
+            const userAnswers = this._prepareSubmitValues(formData, params);
+            var questionIds = $form.find('.js_question-wrapper').map(function() { return this.id; }).get();
+
+            // Show correction before submit if survey scoring type is 'scoring_with_answers_after_page'
+            if (self.options.scoringAfterEachPage && params) {
+                if (self.showAnswers || self.showAnswers === undefined) {
+                    const correctAnswers = await self._rpc({
+                        model: 'survey.question',
+                        method: 'get_correct_answers',
+                        args: [questionIds],
+                    });
+                    // Show answers if there's at least one to display
+                    if (!Object.values(correctAnswers).every(answer => answer === null)) {
```

Comment:
Sadly it doesn't seem to work :slightly_frowning_face: or did I get it wrong ?
![image](https://github.com/odoo/odoo/assets/107679666/065a02eb-1c97-444d-b3c1-63f9454d34cf)
The correct answer is 8 so it should return `true` to display the answer but it returns false.
`8 == true` is evaluated to `false`

------------------------------------------

If we think about it, the `get_correct_answer` function returns those types in js:
`numerical, date` => int or float 
`date` => string 
`datetime` => string 
`simple choice` => list of integers
`multiple choice` => list of integers
![image](https://github.com/odoo/odoo/assets/107679666/cc9a0c82-603b-47c6-922a-8b88d52d0a59)
So maybe using this could work ?:
```js
const invalidAnswer = (answer) => answer == null || (['string', 'object'].includes(typeof answer) && !answer.length);
if (Object.values(correctAnswers).some(answer => !invalidAnswer(answer))) {
   ...
}
```

`answer == null` is true if answer is `undefined` or `null`
`['string', 'object'].includes(typeof answer) && !answer.length` is true if the answer is a string or an object (this includes the array type) and if it is empty

Which means, if it is not `null` or `undefined` or an `empty array, object or string` then it is valid.

What do you think about this? :slightly_smiling_face:

Reply:
If it didn't work that's ok, thought I checked every case but I guess not !
Let's go for this then, it checks every case of what could be returned as an invalid answer and is easier to read.